fluid/PaddleCV/int8/utility.py

```diff
@@ -0,0 +1,336 @@
+    )
+
+    program.current_block().append_op(
+        type = 'save',
+        inputs={'X': '{}'.format(name)},
+        outputs={},
+        attrs={"file_path": "{}/{}".format(dst, name)}
+    )
+
+    save_prog.current_block().create_var(
+        name=name,
+        dtype=data_type,
+        persistable=True,
+    )
+
+def check_op_ancestor(program, var_name):
+    search_end_index = 0
+    input_index_name = {}
+    output_index_name = {}
+    ops_type = []
+    found_var = False
+    first_conv_op_index = -1
+    first_conv_op_flag = False
+    for index, op in enumerate(program.current_block().ops):
+        input_var_list = []
+        output_var_list = []
+        ops_type.append(op.type)
+        if op.type == 'conv2d' and not first_conv_op_flag:
+            first_conv_op_index = index
+            first_conv_op_flag = True
+        op_input_name = op.input_names[0]
+        input_var_list.append(op.input(op_input_name)[0])
+        input_index_name[index] = input_var_list
+        
+        for name in op.output_names:
+            output_var_list.append(op.output(name)[0])
+
+            if op.output(name)[0] == var_name:
+                search_end_index = index
+                found_var = True
+                break
+        output_index_name[index] = output_var_list
+
+        if found_var: break
+    # analysis
+    while search_end_index >= 0:
+        input_name = input_index_name[search_end_index][0]
+        found_ancestor = False
+        for i in output_index_name.keys():
+            if input_name in output_index_name[i]:
+                search_end_index = i
+                found_ancestor = True
+                break
+
+        if not found_ancestor: # Dangling var
+            return S8_MAX
+        elif search_end_index == first_conv_op_index: # first conv
+            if program.current_block().ops[i].has_attr('fuse_relu') and program.current_block().ops[i].attr('fuse_relu'):
+                return U8_MAX
+            else:
+                return S8_MAX
+        elif ops_type[search_end_index] != 'conv2d':
+            continue
+        else:
+            if program.current_block().ops[i].has_attr('fuse_relu') and program.current_block().ops[i].attr('fuse_relu'):
+                return U8_MAX
+            else:
+                return S8_MAX
+    return S8_MAX
+
+
+def update_program_attr(program, attr="is_test", value=True):
+    special_op_type_list = [
+                    "pool2d", "sigmoid", "logsigmoid", "softshrink", "exp",
+                    "brelu", "pow", "leaky_relu", "stanh", "relu", "tanh",
+                    "tanh_shrink", "sqrt", "abs", "ceil", "elu", "floor", "cos",
+                    "sin", "round", "reciprocal", "hard_shrink", "hard_sigmoid",
+                    "relu6", "soft_relu", "swish", "thresholded_relu", "log",
+                    "square", "softplus", "softsign"]
+    for i in program.current_block().ops:
+        if i.has_attr(attr) or i.type in special_op_type_list:
```

Comment:
why do you need this special_op_type_list? Are they behaving differently when `is_test=True`?

Reply:
Will remove.